data/templates/src/pe/dll/build.bat

```diff
@@ -3,12 +3,13 @@
 if "%~1"=="" GOTO NO_ARGUMENTS
 echo Compiling for: %1
 call "%VCINSTALLDIR%Auxiliary\Build\vcvarsall.bat" %1
-cl /LD /GS- /DBUILDMODE=2 template.c /Fe:template_%1_windows.dll /link kernel32.lib /entry:DllMain /subsystem:WINDOWS
+rc /v template.rc
+cl /LD /GS- /DBUILDMODE=2 template.c /Fe:template_%1_windows.dll /link kernel32.lib template.res /entry:DllMain /subsystem:WINDOWS
+cl /LD /GS- /DBUILDMODE=2 /DSCSIZE=262144 template.c /Fe:template_%1_windows.256kib.dll /link kernel32.lib template.res /entry:DllMain /subsystem:WINDOWS
 exit /B
 
 :NO_ARGUMENTS
 %COMSPEC% /c "%0" x86
 %COMSPEC% /c "%0" x64
-del  *.obj
+del  *.obj *.res
```

Comment:
> Confused by this? Your both building the resource files and then deleting them right after they have been built?

Yes, `rc /v template.rc` takes `template.rc` as the input file and generates `template.res` which is then used as the input to `cl` on lines 7 and 8. Once the build is complete, the `.res` file is not necessary, so it's deleted.

> Additionally, looking at the entire build chain further I'm confused why dll_mixed_mode appears to be the only one that doesn't create a resource file using the rc compiler.

`dll_mixed_mode` doesn't use the resource file because it has never used that resource file. `dll_gdiplus` used it but it was consolidated in this PR in 2608852d8cd5d3badac03195b325fb2a887d81cd. The original `dll` one used to use it but it was removed in this commit 2d367b867dd165441b23eb22fcd87568cbd54af7 from PR #14410 when I first introduced the bat build script. I don't recall doing it intentionally.

I could update `dll_mixed_mode` to use the resource script, but that's why it's the only one that doesn't use it now.

Reply:
> Yes, `rc /v template.rc` takes `template.rc` as the input file and generates `template.res` which is then used as the input to `cl` on lines 7 and 8. Once the build is complete, the `.res` file is not necessary, so it's deleted.

Woops looks like I overlooked that and likely made this comment before I fully understood how these files interlinked. Makes sense, no issues there.


> > Additionally, looking at the entire build chain further I'm confused why dll_mixed_mode appears to be the only one that doesn't create a resource file using the rc compiler.
> 
> `dll_mixed_mode` doesn't use the resource file because it has never used that resource file. `dll_gdiplus` used it but it was consolidated in this PR in [2608852](https://github.com/rapid7/metasploit-framework/commit/2608852d8cd5d3badac03195b325fb2a887d81cd). The original `dll` one used to use it but it was removed in this commit [2d367b8](https://github.com/rapid7/metasploit-framework/commit/2d367b867dd165441b23eb22fcd87568cbd54af7) from PR #14410 when I first introduced the bat build script. I don't recall doing it intentionally.
> 
> I could update `dll_mixed_mode` to use the resource script, but that's why it's the only one that doesn't use it now.

Ah makes sense. Thanks for clarifying that. Going to mark this as resolved since I have no outstanding concerns then r.e this.

Reply:
Leaving this as unresolved just so we are aware to come back and look at the `dll_mixed_mode` update, this issue itself is fixed though.